### PR TITLE
Define mainClassName in Micronaut example

### LINKED ISSUE #1495: Micronaut  Example Main class was not found, perhaps you should add a `mainClass` configuration to jib
**Description of the issue**:
Micronaut example not compiling

**Expected behavior**:
create docker image

**Steps to reproduce**:  

./gradlew jibDockerBuild

**Environment**: 

Linux, gralde, JDK 11

 
**`jib-gradle-plugin` Configuration**: 
```groovy
plugins {
    id 'application'
    id 'groovy'
    id 'io.spring.dependency-management' version '1.0.6.RELEASE'
    id 'net.ltgt.apt-idea' version '0.18'
    id 'com.google.cloud.tools.jib' version '1.0.0'
}

version '0.1'
group 'example.micronaut-jib'
sourceCompatibility = 1.8

repositories {
    mavenCentral()
    maven { url 'https://jcenter.bintray.com' }
}

dependencyManagement {
    imports {
        mavenBom 'io.micronaut:micronaut-bom:1.0.3'
    }
}

dependencies {
    annotationProcessor 'io.micronaut:micronaut-inject-java'
    compile 'io.micronaut:micronaut-http-client'
    compile 'io.micronaut:micronaut-http-server-netty'
    compile 'io.micronaut:micronaut-runtime-groovy'
    compile 'io.micronaut:micronaut-inject'
    compile 'io.micronaut:micronaut-runtime'
    compileOnly 'io.micronaut:micronaut-inject-groovy'
    compileOnly 'io.micronaut:micronaut-inject-java'
    runtime 'ch.qos.logback:logback-classic:1.2.3'
    testCompile 'io.micronaut:micronaut-inject-groovy'
    testCompile('org.spockframework:spock-core:1.2-groovy-2.5') {
        exclude group: 'org.codehaus.groovy', module: 'groovy-all'
    }
    testCompile 'io.micronaut:micronaut-inject-java'
}

compileJava.options.compilerArgs += '-parameters'
compileTestJava.options.compilerArgs += '-parameters'
run.main = 'example.micronaut.Application'


```

**Log output**: 

./gradlew jibDockerBuild  --image=test
> Task :jibDockerBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':jibDockerBuild'.
> com.google.cloud.tools.jib.plugins.common.MainClassInferenceException: Main class was not found, perhaps you should add a `mainClass` configuration to jib

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

**Additional Information**:

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #1495.

#1243 added `run.main` to fix `./gradlew run` back then. But I couldn't find any documentation about `run.main`.

OTOH, [the use of `mainClassName` is documented in the Micronaunt doc](https://docs.micronaut.io/latest/guide/index.html#groovyFunctions). It makes the sample work again.

~~I had to add `jib.container.mainClass`. This might be due to #1501.~~ UPDATE: #1501 fixed in Jib 1.0.1.

### CODE REVIEW

**Comment by loosebazooka** [comment]
Should we just read mainClassName from gradle-application plugin? It is a standard plugin.

**Comment by chanseokoh** [comment]
Yeah, I think we should. That is suggested in https://github.com/GoogleContainerTools/jib/issues/1501#issuecomment-466214381. I'll add a comment to `build.gradle`. In addition to that, I think [Jib should be able to infer the main](https://github.com/GoogleContainerTools/jib/issues/1501#issuecomment-466216722).

File: examples/micronaut/build.gradle [hunk 1]
```diff
@@ -39,7 +39,10 @@ dependencies {
...
 compileJava.options.compilerArgs += '-parameters'
 compileTestJava.options.compilerArgs += '-parameters'
-run.main = 'example.micronaut.Application'
+mainClassName = 'example.micronaut.Application'
 
 /** CHANGE THIS TO THE IMAGE YOU WANT TO PUSH TO */
 // jib.to.image = 'gcr.io/PROJECT_ID/micronaut-jib'
+
+// not needed once https://github.com/GoogleContainerTools/jib/issues/1501 is fixed
+jib.container.mainClass = mainClassName
```
Comment (coollog) [inline examples/micronaut/build.gradle]: I believe we can remove this if we upgrade the version to `1.0.1`

Comment (coollog) [reply examples/micronaut/build.gradle]: We still should probably use `mainClassName` directly as part of our main class inference though.

Comment (TadCordle) [reply examples/micronaut/build.gradle]: Oh yeah we have to do the PR to update version numbers, I'll get on that

Comment (chanseokoh) [reply examples/micronaut/build.gradle]: OK, I can wait until then. I think we still need to change `run.main` to `mainClassName` to fix #1495.

**Pull request merged**